src/compute/lower.rs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Defines kernel to extract a lower case of a \[Large\]StringArray
+
+use crate::array::*;
+use crate::{
+    datatypes::DataType,
+    error::{ArrowError, Result},
+};
+
+fn utf8_lower<O: Offset>(array: &Utf8Array<O>) -> Utf8Array<O> {
+    let iter = array.values_iter().map(str::to_lowercase);
+
+    let new = Utf8Array::<O>::from_trusted_len_values_iter(iter);
```

Comment:
originally I thought that an invariant of lower would preserve the number of bytes on the string and thus could be optimized away, but now that I look at this I am not sure it does any more xD

Reply:
Utf8 is so complex that I don't know is it possible.